src/shipping/src/shipping_service/quote.rs

```diff
@@ -50,17 +49,21 @@ async fn request_quote(count: u32) -> Result<f64, Box<dyn std::error::Error>> {
     let mut reqbody = HashMap::new();
     reqbody.insert("numberOfItems", count);
 
-    let client = ClientBuilder::new(reqwest::Client::new())
-        .with(TracingMiddleware::<SpanBackendWithUrl>::new())
-        .build();
-
-    let resp = client
+    let mut response = client
         .post(quote_service_addr)
-        .json(&reqbody)
-        .send()
-        .await?
-        .text_with_charset("utf-8")
-        .await?;
+        .trace_request()
+        .send_json(&reqbody)
+        .await
```

Comment:
You should use `with_current_context` whenever you are going to pass control into a future, otherwise the other side of the await won't be associated with the proper calling context 

https://docs.rs/opentelemetry/latest/opentelemetry/trace/trait.FutureExt.html#method.with_current_context